astropy/io/misc/parquet.py

```diff
@@ -0,0 +1,304 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This package contains functions for reading and writing Parquet
+tables that are not meant to be used directly, but instead are
+available as readers/writers in `astropy.table`.  See
+:ref:`astropy:table_io` for more details.
+"""
+
+import os
+import warnings
+
+import numpy as np
+
+# NOTE: Do not import anything from astropy.table here.
+# https://github.com/astropy/astropy/issues/6604
+from astropy.utils.exceptions import AstropyUserWarning
+
+PARQUET_SIGNATURE = b'PAR1'
+
+__all__ = ['read_table_parquet', 'write_table_parquet']
+
+
+def is_parquet(origin, filepath, fileobj, *args, **kwargs):
+
+    if fileobj is not None:
+        pos = fileobj.tell()
```

Comment:
This will fail if ``fileobj`` is not None but the object is not the correct type.
Maybe use a ``try: except:`` to ensure ``is_parquet`` always returns a bool.

Reply:
Thanks!